stdlib/src/builtin/string.mojo

```diff
@@ -2109,3 +2198,76 @@ fn _calc_format_buffer_size[type: DType]() -> Int:
         return 64 + 1
     else:
         return 128 + 1  # Add 1 for the terminator
+
+
+# ===----------------------------------------------------------------------===#
+# Format method structures
+# ===----------------------------------------------------------------------===#
+
+
+@value
+struct _FormatCurlyEntry:
+    """
+    Internally used by the `format()` method.
+
+    Purpose of the variant types:
+
+
+    - `Int` for manual indexing of *args
+
+        ```mojo
+        "{0}".format("hello")
+        ```
+
+        value field contains `0`
```

Comment:
It would be better to move this description to the `create_entries` function, since that's where this logic actually lives. Also, just to decrease cognitive load, let's use an example that doesn't refer to `format`, e.g. describing the result of
```mojo
_FormatCurlyEntry.create_entries("{0} foo {1}")
```